program-runtime/src/compute_budget_processor.rs

```diff
@@ -616,4 +627,78 @@ mod tests {
             })
         );
     }
+
+    fn try_prioritization_fee_from_deprecated_requested_units(
+        additional_fee: u32,
+        compute_unit_limit: u32,
+    ) {
+        let payer_keypair = Keypair::new();
+        let tx = SanitizedTransaction::from_transaction_for_tests(Transaction::new(
+            &[&payer_keypair],
+            Message::new(
+                &[Instruction::new_with_borsh(
+                    compute_budget::id(),
+                    &compute_budget::ComputeBudgetInstruction::RequestUnitsDeprecated {
+                        units: compute_unit_limit,
+                        additional_fee,
+                    },
+                    vec![],
+                )],
+                Some(&payer_keypair.pubkey()),
+            ),
+            Hash::default(),
+        ));
+
+        // sucessfully process deprecated instruction
+        let compute_budget_limits = process_compute_budget_instructions(
+            tx.message().program_instructions_iter(),
+            &FeatureSet::default(),
+        )
+        .unwrap();
+
+        // assert compute_budget_limit
+        let expected_compute_unit_price = (additional_fee as u128)
+            .saturating_mul(1_000_000)
+            .checked_div(compute_unit_limit as u128)
+            .map(|cu_price| u64::try_from(cu_price).unwrap_or(u64::MAX))
+            .unwrap();
+        let expected_compute_unit_limit = compute_unit_limit.min(MAX_COMPUTE_UNIT_LIMIT);
+        assert_eq!(
+            compute_budget_limits.compute_unit_price,
+            expected_compute_unit_price
+        );
+        assert_eq!(
+            compute_budget_limits.compute_unit_limit,
+            expected_compute_unit_limit
+        );
+
+        // assert fee_budget_limits
+        let fee_budget_limits = FeeBudgetLimits::from(compute_budget_limits);
+        assert_eq!(
+            fee_budget_limits.prioritization_fee,
+            u64::from(additional_fee)
+        );
+        assert_eq!(
+            fee_budget_limits.compute_unit_limit,
+            u64::from(expected_compute_unit_limit)
+        );
+    }
+
+    #[test]
+    fn test_support_deprecated_requested_units() {
```

Comment:
To confirm, does this unit test fail before the addition of `deprecated_additional_fee`? I don't think it is necessary to retroactively split the commits on this PR, but in general, it can be very nice to break things up into two+ commits:
- The first commit that introduces a test that fails with current state (typically master or in this PR's case, the commits from the initial PR)
- The subsequent commit(s) that address the issue (ie get the unit test passing)

Again, I don't think strictly necessary to split the commits for this PR, but maybe you can tweak things locally to run the unit test without the fix and confirm you see the failure as an extra sanity check. The fact that you see the correct bank hash on the mnb ledger seemingly indicates you have resolved things, but hopefully this is something you can run quickly that will give us an extra sanity check

Reply:
Yep, I added this test first to confirm the theory; Ran it and confirmed it failed with exact priority fee as I got from ledge-tool replay that resulted mismatching hash. Then I did the fix (adding optional `deprecated_additional_fee`) to green the test. 

I agree that separate commits are better. In this case, I thought appending one self-contained commit to already reviewed changes would reduce cluster, so I did `git rebase -i HEAD~2`. Look back, keeping 2 commits ain't that bad at all :)

Reply:
Good deal / thanks for confirming!